src/lib/lib.jl

```diff
@@ -2,12 +2,12 @@ using Base: RefValue
 
 # Interfaces
 
-accum() = nothing
+accum() = Zero()
 accum(x) = x
 
 accum(x, y) =
-  x === nothing ? y :
-  y === nothing ? x :
+  x isa Union{Nothing, AbstractZero} ? y :
+  y isa Union{Nothing, AbstractZero} ? x :
   x + y
```

Comment:
We don't need this change because `+` on a`AbstactZero` works.
so can leave this as `nothing`

Reply:
So I thought but this fixed the overloading `+(NamedTuple, NamedTuple)` error for nested AD. Essentially there are two `Tuple{Zero,NamedTuple{(:back,),Tuple{NamedTuple{(:back,),Tuple{NamedTuple{(:x,),Tuple{Float64}}}}}}}` being added for which `accum` mysteriously breaks. It works if I define it in the `REPL`. It feels like the `@generated accum(NT, NT)` should be called but it isn't. It's really weird.

Reply:
And yes it smells like a bug I should have added a comment